src/ros_can_nodes/scripts/generate_message_properties_map.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python
+
+import sys
+import itertools
+import rosmsg
+
+try:
+    from cStringIO import StringIO #Python 2.x
+except ImportError:
+    from io import StringIO #Python 3.x
+
+
+hpp_contents =\
+"""
+/* Auto-generated for ros_can_nodes */
+/* DO NOT EDIT */
+#ifndef MESSAGE_PROPERTIES_MAP_HPP
+#define MESSAGE_PROPERTIES_MAP_HPP
+
+#include <unordered_map>
+#include <string>
+
+struct message_properties {
+    const std::string definition;
+    const std::string md5sum;
+};
+
+extern const std::unordered_map<std::string, const message_properties> message_properties_map;
+
+#endif // MESSAGE_PROPERTIES_MAP_HPP
+"""
+
+cpp_contents =\
+"""
+/* Auto-generated for ros_can_nodes */
+/* DO NOT EDIT */
+
+#include <ros/message_traits.h>
+#include <unordered_map>
+#include <string>
+#include "message_properties_map.hpp"
+
+{0}
+
+const std::unordered_map<std::string, const message_properties> message_properties_map = {{
+{1}
+}};
+"""
+
+map_entry_template =\
+'{{ std::string{{ros::message_traits::DataType<{0}>::value()}},' +\
+'{{ std::string{{ros::message_traits::Definition<{0}>::value()}},' +\
+'std::string{{ros::message_traits::MD5Sum<{0}>::value()}} }} }},'
```

Comment:
rather than having to depend on all the c header files (and having all the messages being generated) you could use the values for the python library here